powertools-serialization/src/main/java/software/amazon/lambda/powertools/utilities/EventDeserializer.java

```diff
@@ -0,0 +1,150 @@
+        } else if (obj instanceof KinesisAnalyticsStreamsInputPreprocessingEvent) {
+            KinesisAnalyticsStreamsInputPreprocessingEvent event = (KinesisAnalyticsStreamsInputPreprocessingEvent) obj;
+            return new EventPart(event.getRecords().stream().map(r -> decode(r.getData())).collect(Collectors.toList()));
+        } else {
+            // does not really make sense to use this EventLoader when you already have a typed object
+            // just not to throw an exception
+            LOG.warn("Consider using your object directly instead of using EventDeserializer");
+            return new EventPart(obj);
+        }
+    }
+
+    public static class EventPart {
+        private Map<String, Object> contentMap;
+        private String content;
+        private List<String> contentList;
+        private Object contentObject;
+
+        public EventPart(List<String> contentList) {
+            this.contentList = contentList;
+        }
+
+        public EventPart(String content) {
+            this.content = content;
+        }
+
+        public EventPart(Map<String, Object> contentMap) {
+            this.contentMap = contentMap;
+        }
+
+        public EventPart(Object content) {
+            this.contentObject = content;
+        }
+
+        public <T> T as(Class<T> clazz) {
+            try {
+                if (content != null) {
+                    if (content.getClass().equals(clazz)) {
+                        // do not read json when returning String, just return the String
+                        return (T) content;
+                    }
+                    return JsonConfig.get().getObjectMapper().reader().readValue(content, clazz);
+                }
+                if (contentMap != null) {
+                    return JsonConfig.get().getObjectMapper().convertValue(contentMap, clazz);
+                }
+                if (contentObject != null) {
+                    return (T) contentObject;
+                }
+                if (contentList != null) {
+                    throw new EventDeserializationException("The content of this event is a list, consider using 'extractDataAsListOf' instead");
+                }
+                throw new EventDeserializationException("Event content is null");
+            } catch (IOException e) {
+                throw new EventDeserializationException("Cannot load the event as " + clazz.getSimpleName(), e);
+            }
+        }
+
+        public <T> List<T> asListOf(Class<T> clazz) {
+            if (contentList == null) {
+                throw new EventDeserializationException("Event content is null");
+            }
+            return contentList.stream().map(s -> {
+                try {
+                    return JsonConfig.get().getObjectMapper().reader().readValue(s, clazz);
+                } catch (IOException e) {
+                    throw new EventDeserializationException("Cannot load the event as " + clazz.getSimpleName(), e);
```

Comment:
Possible to validate this code path in tests? Seems missing

Reply:
Yes